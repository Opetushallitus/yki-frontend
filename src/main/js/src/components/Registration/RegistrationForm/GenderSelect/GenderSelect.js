import React from 'react';
import { Field } from 'formik';
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';
import * as R from 'ramda';

export const genderSelect = props => {
  const sortByCode = R.sortBy(R.prop('koodiArvo'));
  const gendersByLocale = sortByCode(props.genders).map(g => {
    const metadata = g.metadata.find(
      m => m.kieli === props.i18n.language.toUpperCase(),
    );
    return (
      <option value={g.koodiArvo} key={g.koodiArvo}>
        {metadata.nimi}
      </option>
    );
  });

  return (
    <React.Fragment>
      <h3>{props.t('registration.form.gender')}</h3>
      <Field
        component="select"
        name="gender"
        className={props.className}
        data-cy="select-gender"
      >
        <option value="" key="" />
        {gendersByLocale}
      </Field>
    </React.Fragment>
  );
};

genderSelect.propTypes = {
  genders: PropTypes.array.isRequired,
  className: PropTypes.string.isRequired,
};

export default withTranslation()(genderSelect);
