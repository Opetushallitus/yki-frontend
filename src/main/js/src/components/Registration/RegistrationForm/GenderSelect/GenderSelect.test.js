import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { genderSelect as GenderSelect } from './GenderSelect';

configure({ adapter: new Adapter() });

describe('<GenderSelect />', () => {
  it('should render gender selection sorted by code', () => {
    const genders = [
      {
        koodiArvo: '2',
        metadata: [
          {
            nimi: 'nainen',
            kieli: 'FI',
          },
        ],
      },
      {
        koodiArvo: '1',
        metadata: [
          {
            nimi: 'mies',
            kieli: 'FI',
          },
        ],
      },
    ];

    const wrapper = shallow(
      <GenderSelect
        t={key => key}
        lng={'fi'}
        genders={genders}
        className="test"
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
