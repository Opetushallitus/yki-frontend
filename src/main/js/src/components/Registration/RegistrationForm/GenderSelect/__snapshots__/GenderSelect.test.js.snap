// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenderSelect /> should render gender selection sorted by code 1`] = `
<div
  className="InputFieldWrapper"
>
  <label
    htmlFor="gender-select"
  >
    registration.form.gender
  </label>
  <FormikConnect(FieldInner)
    aria-required={true}
    className="test"
    component="select"
    data-cy="select-gender"
    id="gender-select"
    name="gender"
  >
    <option
      key="placeholder"
      value="placeholder"
    >
      common.selectorDefault
    </option>
    <option
      key="1"
      value="1"
    >
      mies
    </option>
    <option
      key="2"
      value="2"
    >
      nainen
    </option>
  </FormikConnect(FieldInner)>
</div>
`;
